contracts/Pool.sol

```diff
@@ -129,19 +135,21 @@ contract MolochPool {
     function _mintSharesForAddress(uint256 sharesToMint, address recipient) internal {
         totalPoolShares = totalPoolShares.add(sharesToMint);
         donors[recipient].shares = donors[recipient].shares.add(sharesToMint);
+
+        require(totalPoolShares <= MAX_NUMBER_OF_SHARES);
     }
 
-    function _withdraw(address recipient, uint256 sharesToBurn) {
-        Donor donor = donors[recipient];
+    function _withdraw(address recipient, uint256 sharesToBurn) internal {
+        Donor storage donor = donors[recipient];
 
-        require(donors[recipient].shares >= sharesToBurn);
+        require(donor.shares >= sharesToBurn);
 
         uint256 tokensToWithdraw = approvedToken.balanceOf(address(this)).mul(sharesToBurn).div(totalPoolShares);
 
         totalPoolShares = totalPoolShares.sub(sharesToBurn);
-        donors[msg.sender].shares = donors[msg.sender].shares.sub(sharesToBurn);
+        donor.shares = donor.shares.sub(sharesToBurn);
 
-        require(approvedToken.transfer(msg.sender, tokensToWithdraw));
+        require(approvedToken.transfer(recipient, tokensToWithdraw));
```

Comment:
This changed the behavior of `keeperWithdraw`, and now the money always goes to the donor. Not sure if this is intended. If not, a simpler solution can be to add an extra param `donor`.

Reply:
That's right. I was assuming that all occurrences of `msg.sender` in the (now internal) `_withdraw` function were erroneous leftovers from the public version before the introduction of keepers. But indeed, your concern is valid.
@ameensol, can you clarify the role of the keepers: For a `keeperWithdraw`, who should be the receiver of the transfer: the keeper or the original donor?

Reply:
The keeper should never be the recipient of the funds, they are simply there to watch the contract and in case Moloch is about to pass a malicious proposal (e.g to transfer Guild Bank funds to a member thus also directing Pool funds to the member), to execute `keeperWithdraw` and return the money to the donor.

Reply:
Thanks. That's what the PR does.